README.md
[![Coffee IT - Android Aroma BottomSheet Component](https://coffeeit.nl/wp-content/uploads/2022/09/Aroma-bottom-sheet-cover-Android-1.png)](https://coffeeit.nl/)

[![Maven Central](https://maven-badges.herokuapp.com/maven-central/nl.coffeeit.aroma/bottomsheet/badge.svg)](https://maven-badges.herokuapp.com/Maven-Central/nl.coffeeit.aroma/bottomsheet)
[![API](https://img.shields.io/badge/API-21%2B-brightgreen.svg?style=flat)](https://android-arsenal.com/api?level=21)
[![License](https://img.shields.io/badge/license-MIT-brightgreen.svg)](https://github.com/Coffee-IT-Development/BottomSheet-Android-Component/blob/main/LICENSE)
[![Mirror Repository](https://img.shields.io/badge/Mirror-Repository-9b34eb?style=flat-square)](https://github.com/Coffee-IT-Development/BottomSheet-Android-Component)
[![LinkedIn](https://img.shields.io/badge/LinkedIn-@CoffeeIT-blue.svg?style=flat-square)](https://linkedin.com/company/coffee-it)
[![Facebook](https://img.shields.io/badge/Facebook-CoffeeITNL-blue.svg?style=flat-square)](https://www.facebook.com/CoffeeITNL/)
[![Instagram](https://img.shields.io/badge/Instagram-CoffeeITNL-blue.svg?style=flat-square)](https://www.instagram.com/coffeeitnl/)
[![Follow coffeeitnl on Twitter](https://img.shields.io/twitter/follow/coffeeitnl.svg?style=flat-square&logo=twitter)](https://twitter.com/coffeeitnl)

# ☕️ Android Aroma BottomSheet

The Android Aroma BottomSheet package provides a customisable BottomSheet. This README describes how to implement the BottomSheet into an app.

This GitHub repository is a mirror, the official repository is hosted privately by Coffee IT.

Created by [Coffee IT](https://coffeeit.nl/).

<img src="docs/images/Demo.gif" width="300">

# ⚡ Installation

This component requires minimum SDK 21.

Add the dependency to `build.gradle`.

Groovy:
```Groovy
dependencies {
    implementation "nl.coffeeit.aroma:bottomsheet:1.0.3"
}
```

KTS:
```Kotlin
dependencies {
    implementation("nl.coffeeit.aroma:bottomsheet:1.0.3")
}
```

# 📖 Usage

To use the BottomSheet in a project, simply add it to your code like this:

```Kotlin
BottomSheetWithContent(
    activityContentScope = { state, scope ->
        Button(onClick = {
            scope.launch {
                state.show()
            }
        }) {
            Text("This goes behind the sheet")
        }
    },
    
    sheetContent = {
        Text(text = "This goes inside the sheet")
    }
)
```
Two parameters are required:

`activityContentScope` holds the content that should be displayed initially, when the bottom sheet is hidden. It offers a `state` to toggle the visibility of the bottom sheet and a coroutine `scope`

`sheetContent` represents a Composable that is the the content of the actual sheet. XML can also be used here, since it can be inflated in a Composable.

# ⚙️ Customisation

`BottomSheetWithContent` accepts the following optional parameters:

- `backgroundColor` A `Color` object to set the background of the bottom sheet
- `scrimColor` A `Color` object to set the color of the area behind the sheet
- `width` The width of the bottom sheet (default is fill screen width)
- `bottomMargin` Margin for the bottom of the sheet, to make a floating variant (default 0)
- `cornerShape` A `RoundedCornerShape` object to give the sheet rounded edges
- `accessory` Decides the accessory of the BottomSheet, can be:
  -- `NONE`
  -- `GRABBER`
  -- `CLOSE_BUTTON`

# ✏️ Changelog

The changelog can be found [here](https://github.com/Coffee-IT-Development/BottomSheet-Android-Component/blob/main/CHANGELOG.md).

# 🔗 Related publications

- [Aroma BottomSheet Component for iOS](https://github.com/Coffee-IT-Development/BottomSheetModal-iOS-Component)

Look at our other repositories on our [GitHub account](https://github.com/orgs/Coffee-IT-Development/repositories).

# 📧 Contact

Do you have questions, ideas or need help? Send us an email at contact@coffeeit.nl.

<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://global-uploads.webflow.com/605a171ee93af49275331843/623b23cdea80a92703e61b42_Logo_black_1.svg" width="100">
  <source media="(prefers-color-scheme: light)" srcset="https://coffeeit.nl/wp-content/uploads/2016/09/logo_dark_small_new.png" width="100">
  <img alt="The Coffee IT logo" src="https://coffeeit.nl/wp-content/uploads/2016/09/logo_dark_small_new.png">
</picture>

# ⚠️ License

Android Aroma BottomSheet is licensed under the terms of the [MIT Open Source license](https://github.com/Coffee-IT-Development/BottomSheet-Android-Component/blob/main/LICENSE).

CHANGELOG.md
# Change log

Version 1.0.3 *(22-09-2022)*
----------------------------
* Migrated from `compileSdk` and `targetSdk` 32 to 33.

Version 1.0.2 *(07-09-2022)*
----------------------------
* Removed build variants because it caused issues in other projects.

Version 1.0.1 *(31-08-2022)*
----------------------------
* Renamed `bottomPadding` to `bottomMargin`.
* Replaced close icon.
* Fixed close button ripple.

Version 1.0.0 *(01-07-2022)*
----------------------------
* Added BottomSheet component.

bottomsheet/src/main/java/nl/coffeeit/aroma/bottomsheet/Grabber.kt
/*
 * Created by Coffee IT
 *
 * MIT License
 *
 * Copyright (c) 2022 Coffee IT
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package nl.coffeeit.aroma.bottomsheet

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp


@Composable
internal fun Grabber(modifier: Modifier) {
    Column(
        modifier = modifier
            .height(4.dp)
            .width(45.dp)
            .wrapContentSize()
    ) {
        Box(
            modifier = Modifier
                .size(100.dp)
                .clip(RoundedCornerShape(8.dp))
                .background(Color(0xFFB4B4B4))
        )
    }
}

bottomsheet/src/main/java/nl/coffeeit/aroma/bottomsheet/Accessory.kt
/*
 * Created by Coffee IT
 *
 * MIT License
 *
 * Copyright (c) 2022 Coffee IT
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package nl.coffeeit.aroma.bottomsheet

enum class Accessory(
    val displayName: String
) {
    NONE("None"), GRABBER("Grabber"), CLOSE_BUTTON("Close button")
}

bottomsheet/src/main/java/nl/coffeeit/aroma/bottomsheet/BottomSheet.kt
/*
 * Created by Coffee IT
 *
 * MIT License
 *
 * Copyright (c) 2022 Coffee IT
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package nl.coffeeit.aroma.bottomsheet

import androidx.compose.foundation.Image
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CornerSize
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.ui.Alignment
import androidx.compose.ui.Alignment.Companion.CenterHorizontally
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.launch

private const val DEFAULT_BACKGROUND_COLOR = 0xFFF6F6F6
private const val DEFAULT_CORNER_RADIUS = 16
private const val DEFAULT_SCRIM_COLOR = 0x99000000

@Composable
@ExperimentalMaterialApi
fun BottomSheetWithContent(
    activityContentScope: @Composable (state: ModalBottomSheetState, scope: CoroutineScope) -> Unit,
    sheetContent: @Composable () -> Unit,
    backgroundColor: Color = Color(DEFAULT_BACKGROUND_COLOR),
    scrimColor: Color = Color(DEFAULT_SCRIM_COLOR),
    width: Float? = null,
    bottomMargin: Float = 0f,
    cornerShape: RoundedCornerShape = RoundedCornerShape(DEFAULT_CORNER_RADIUS.dp),
    accessory: Accessory = Accessory.NONE
) {

    val state = rememberModalBottomSheetState(
        initialValue = ModalBottomSheetValue.Hidden
    )
    val scope = rememberCoroutineScope()

    val sheetSurfaceModifier = if (width != null && width > 0) {
        Modifier.width(width.dp)
    } else {
        Modifier.fillMaxWidth()
    }

    ModalBottomSheetLayout(
        sheetState = state,
        scrimColor = scrimColor,
        sheetElevation = 0.dp,
        sheetBackgroundColor = Color.Transparent,
        sheetContent = {
            Surface(
                modifier = sheetSurfaceModifier
                    .align(CenterHorizontally)
                    .padding(bottom = bottomMargin.dp),
                color = backgroundColor,
                shape = if (bottomMargin != 0f) {
                    RoundedCornerShape(
                        cornerShape.topStart,
                        cornerShape.topEnd,
                        cornerShape.bottomStart,
                        cornerShape.bottomEnd
                    )
                } else {
                    RoundedCornerShape(
                        cornerShape.topStart,
                        cornerShape.topEnd,
                        CornerSize(0.dp),
                        CornerSize(0.dp)
                    )
                }
            ) {

                Column {
                    when (accessory) {
                        Accessory.GRABBER -> {
                            Grabber(
                                modifier = Modifier
                                    .padding(top = 12.dp, bottom = 40.dp)
                                    .align(CenterHorizontally)
                            )
                        }
                        Accessory.CLOSE_BUTTON -> {
                            Box(
                                modifier = Modifier
                                    .wrapContentHeight()
                                    .align(Alignment.End)
                                    .padding(16.dp)
                            ) {
                                Box(
                                    modifier = Modifier
                                        .clip(RoundedCornerShape(100.dp))
                                        .clickable {
                                            scope.launch { state.hide() }
                                        }
                                ) {
                                    Image(
                                        painter = painterResource(id = R.drawable.ic_close),
                                        contentDescription = accessory.displayName
                                    )
                                }
                            }
                        }
                        Accessory.NONE -> {
                            // Nothing needs to happen
                        }
                    }
                    sheetContent()
                }
            }
        }
    ) {
        activityContentScope(state, scope)
    }
}



sample/src/main/java/nl/coffeeit/sample/MainActivity.kt
/*
 * Created by Coffee IT
 *
 * MIT License
 *
 * Copyright (c) 2022 Coffee IT
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package nl.coffeeit.sample

import android.annotation.SuppressLint
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.BackHandler
import androidx.activity.compose.setContent
import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.selection.selectable
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.github.ajalt.colormath.model.HSV
import com.github.ajalt.colormath.model.RGB
import com.godaddy.android.colorpicker.harmony.ColorHarmonyMode
import com.godaddy.android.colorpicker.harmony.HarmonyColorPicker
import kotlinx.coroutines.launch
import nl.coffeeit.aroma.bottomsheet.Accessory
import nl.coffeeit.aroma.bottomsheet.BottomSheetWithContent
import java.math.BigDecimal
import kotlin.math.roundToInt

const val DEFAULT_CORNER_RADIUS = 16f
const val DEFAULT_SCRIM_COLOR = 0x99B0B0B3
const val DEFAULT_SCRIM_COLOR_ALPHA = 0.7
const val DEFAULT_SCRIM_COLOR_BLUE = 176
const val DEFAULT_SCRIM_COLOR_GREEN = 176
const val DEFAULT_SCRIM_COLOR_RED = 153

class MainActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            ComponentsDemoScreen()
        }
    }
}

@OptIn(ExperimentalMaterialApi::class)
@SuppressLint("UnusedMaterialScaffoldPaddingParameter")
@Composable
fun ComponentsDemoScreen(
) {
    var bottomSheetAccessory by remember { mutableStateOf(Accessory.NONE) }
    var bottomSheetBottomMargin by remember { mutableStateOf(0f) }
    var bottomSheetRgbColor by remember {
        mutableStateOf(
            RGB(
                DEFAULT_SCRIM_COLOR_RED,
                DEFAULT_SCRIM_COLOR_GREEN,
                DEFAULT_SCRIM_COLOR_BLUE,
                DEFAULT_SCRIM_COLOR_ALPHA
            )
        )
    }
    var bottomSheetRoundedCornerSize by remember { mutableStateOf(DEFAULT_CORNER_RADIUS) }
    var bottomSheetScrimColor by remember { mutableStateOf(Color(DEFAULT_SCRIM_COLOR)) }
    var bottomSheetScrimColorAlpha by remember { mutableStateOf(DEFAULT_SCRIM_COLOR_ALPHA) }
    var bottomSheetWidth by remember { mutableStateOf(0f) }

    val accessoryItems = listOf(Accessory.NONE, Accessory.GRABBER, Accessory.CLOSE_BUTTON)
    val selectedAccessory = remember { mutableStateOf(Accessory.NONE) }
    val isAccessorySelected: (Accessory) -> Boolean = { selectedAccessory.value == it }
    val onAccessoryStateChange: (Accessory) -> Unit = { selectedAccessory.value = it }

    BottomSheetWithContent(
        { state, scope ->
            Column {
                Spacer(modifier = Modifier.height(48.dp))

                Column(
                    Modifier
                        .padding(horizontal = 16.dp)
                ) {
                    Column(
                        modifier = Modifier
                            .verticalScroll(rememberScrollState())
                    ) {

                        Column {
                            Text(text = "Corner radius: ${bottomSheetRoundedCornerSize.toInt()}dp")

                            Slider(value = bottomSheetRoundedCornerSize, onValueChange = { value ->
                                val roundedValue = value.roundToInt().toFloat()
                                bottomSheetRoundedCornerSize = roundedValue
                            }, valueRange = 0f..72f)

                            Spacer(modifier = Modifier.width(32.dp))

                            Text(text = "Width: ${bottomSheetWidth.toInt()}dp (enter 0dp for full width)")

                            Slider(value = bottomSheetWidth, onValueChange = { value ->
                                val roundedValue = value.roundToInt().toFloat()
                                bottomSheetWidth = roundedValue
                            }, valueRange = 0f..250f)

                            Spacer(modifier = Modifier.width(32.dp))

                            Text(text = "Bottom margin: ${bottomSheetBottomMargin.toInt()}dp")

                            Slider(value = bottomSheetBottomMargin, onValueChange = { value ->
                                val roundedValue = value.roundToInt().toFloat()
                                bottomSheetBottomMargin = roundedValue
                            }, valueRange = 0f..250f)
                        }

                        Spacer(modifier = Modifier.height(16.dp))

                        Column {
                            Text(
                                text = "Accessory:"
                            )

                            Row(
                                Modifier
                                    .wrapContentSize()
                                    .padding(top = 8.dp)
                            ) {
                                accessoryItems.forEach { item ->
                                    Row(
                                        verticalAlignment = Alignment.CenterVertically,
                                        modifier = Modifier
                                            .selectable(
                                                selected = isAccessorySelected(item),
                                                onClick = {
                                                    onAccessoryStateChange(item)
                                                    bottomSheetAccessory = item
                                                },
                                                role = Role.RadioButton
                                            )
                                            .padding(end = 4.dp)
                                            .wrapContentSize()
                                    ) {
                                        RadioButton(
                                            selected = isAccessorySelected(item),
                                            onClick = null
                                        )

                                        Spacer(modifier = Modifier.width(4.dp))

                                        Text(
                                            text = item.displayName
                                        )
                                    }
                                }
                            }

                        }

                        Spacer(modifier = Modifier.height(16.dp))

                        Text(text = "Scrim alpha: $bottomSheetScrimColorAlpha")

                        Slider(
                            value = bottomSheetScrimColorAlpha.toFloat(),
                            onValueChange = { value ->
                                var valueFormatted = BigDecimal(value.toDouble())
                                valueFormatted = valueFormatted.setScale(2, BigDecimal.ROUND_UP)
                                bottomSheetScrimColorAlpha = valueFormatted.toDouble()
                                bottomSheetRgbColor = RGB(
                                    bottomSheetRgbColor.r,
                                    bottomSheetRgbColor.g,
                                    bottomSheetRgbColor.b,
                                    bottomSheetScrimColorAlpha
                                )
                                bottomSheetScrimColor = Color(
                                    bottomSheetRgbColor.redInt,
                                    bottomSheetRgbColor.greenInt,
                                    bottomSheetRgbColor.blueInt,
                                    bottomSheetRgbColor.alphaInt
                                )
                            },
                            valueRange = 0f..1f
                        )

                        Spacer(modifier = Modifier.height(16.dp))

                        Text(text = "Scrim color and brightness:")

                        HarmonyColorPicker(
                            harmonyMode = ColorHarmonyMode.SHADES,
                            modifier = Modifier
                                .size(275.dp)
                                .align(Alignment.CenterHorizontally),
                            color = Color(DEFAULT_SCRIM_COLOR),
                            onColorChanged = { hsvColor ->
                                bottomSheetRgbColor = HSV(
                                    hsvColor.hue,
                                    hsvColor.saturation,
                                    hsvColor.value,
                                    bottomSheetScrimColorAlpha
                                ).toSRGB()
                                bottomSheetScrimColor = Color(
                                    bottomSheetRgbColor.redInt,
                                    bottomSheetRgbColor.greenInt,
                                    bottomSheetRgbColor.blueInt,
                                    bottomSheetRgbColor.alphaInt
                                )
                            })

                        Button(modifier = Modifier
                            .fillMaxWidth(),
                            onClick = {
                                scope.launch {
                                    state.show()
                                }
                            }
                        ) {
                            Text(text = "Show BottomSheet")
                        }
                    }
                }

                BackHandler(
                    enabled = (state.currentValue != ModalBottomSheetValue.Hidden),
                    onBack = {
                        scope.launch { state.hide() }
                    }
                )
            }
        },
        sheetContent = {
            Column(
                Modifier
                    .fillMaxWidth()
            ) {

                Image(
                    modifier = Modifier
                        .width(231.dp)
                        .wrapContentHeight()
                        .align(Alignment.CenterHorizontally)
                        .padding(horizontal = 16.dp)
                        .padding(vertical = 72.dp),
                    painter = painterResource(R.drawable.ill_aroma_logo),
                    contentDescription = "Coffee IT's Aroma logo",
                    contentScale = ContentScale.FillWidth
                )
            }
        },
        width = bottomSheetWidth,
        bottomMargin = bottomSheetBottomMargin,
        accessory = bottomSheetAccessory,
        cornerShape = RoundedCornerShape(bottomSheetRoundedCornerSize.dp),
        scrimColor = bottomSheetScrimColor
    )
}


@Composable
@Preview(showBackground = true)
fun ModalBottomSheetHolderPreview() {
    ComponentsDemoScreen()
}